### PR TITLE
Updated tests for user controller

### PR DESCRIPTION
**Pull request by Tallisble**

### CODE REVIEW

File: src/ServiceHub.User.Testing/Service/UserControllerTests.cs [hunk 1]
```diff
@@ -201,18 +239,34 @@ public async Task GetByType_ValidTypes_Return200AndList(string type)
 
             UserController controller = new UserController(mockRepo.Object, new LoggerFactory(), null);
 
-            ObjectResult result = await controller.GetByType(type) as ObjectResult;
-            List<ServiceHub.User.Library.Models.User> users = (List<ServiceHub.User.Library.Models.User>) result?.Value;
+            var result = await controller.GetByType(type);
+            var objectresult = Assert.IsAssignableFrom<ObjectResult>(result);
+            var userobject = Assert.IsAssignableFrom<List<User.Library.Models.User>>(objectresult.Value);
+
+            Assert.Equal(contextUsers.Count, userobject.Count());
```
Comment (fredbelotte) [inline src/ServiceHub.User.Testing/Service/UserControllerTests.cs]: Code Smell Code Smell: Use 'Count' property here instead. ([csharpsquid:S2971](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS2971))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Auser&pullRequest=98&issues=AWVioARYmu5xFVIwHg6V&open=AWVioARYmu5xFVIwHg6V)

[ISSUE_KEY]: AWVioARYmu5xFVIwHg6V

File: src/ServiceHub.User.Testing/Service/UserControllerTests.cs [hunk 2]
```diff
@@ -109,37 +108,74 @@ public async Task InvalidGet_ShouldReturn500()
...
+        [Fact]
+        [Trait("Type", "Controller")]
+        public async Task GetUsersByID_Returns404()
+        {
+            var mockRepo = new Mock<IUserRepository>();
+
+            mockRepo.Setup(x => x.GetByIdAsync(It.IsAny<Guid>())).ReturnsAsync(nullUser1);
+
+            UserController c = new UserController(mockRepo.Object, new LoggerFactory(), null);
+            var result = await c.Get(new Guid());
```
Comment (fredbelotte) [inline src/ServiceHub.User.Testing/Service/UserControllerTests.cs]: Code Smell Code Smell: Use 'Guid.NewGuid()' or 'Guid.Empty' or add arguments to this Guid instantiation. ([csharpsquid:S4581](https://sonarcloud.io/organizations/revaturecloud/rules#rule_key=csharpsquid%3AS4581))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=servicehub%3Auser&pullRequest=98&issues=AWVioARYmu5xFVIwHg6U&open=AWVioARYmu5xFVIwHg6U)

[ISSUE_KEY]: AWVioARYmu5xFVIwHg6U

File: src/ServiceHub.User.Testing/Service/UserControllerTests.cs [hunk 3]
```diff
@@ -15,9 +15,10 @@ namespace ServiceHub.User.Testing.Service
     public class UserControllerTests
     {
         private readonly List<User.Context.Models.User> contextUsers;
+        private readonly User.Context.Models.User nullUser1 = null;
```
Comment (escalonn) [inline src/ServiceHub.User.Testing/Service/UserControllerTests.cs]: no need for this field, just put `null` directly

File: src/ServiceHub.User.Testing/Service/UserControllerTests.cs [hunk 4]
```diff
@@ -109,37 +106,74 @@ public async Task InvalidGet_ShouldReturn500()
...
+        [Fact]
+        [Trait("Type", "Controller")]
+        public async Task ValidGetById_Return200_And_Model()
+        {
+            var mockRepo = new Mock<IUserRepository>();
+            mockRepo.Setup(x => x.GetByIdAsync(It.IsAny<Guid>())).ReturnsAsync(((Guid y) => contextUsers.First(z => z.UserId == y)));
+
+            UserController c = new UserController(mockRepo.Object, new LoggerFactory(), null);
+
+            var result = await c.Get(new Guid("33333333-3333-3333-3333-333333333333"));
```
Comment (escalonn) [inline src/ServiceHub.User.Testing/Service/UserControllerTests.cs]: don't hard-code this twice - store it on the class somewhere

**Pull request merged**